### ISSUE TITLE
ML

### ISSUE THREAD

**Issue by homesh33p** [homesh33p/Switch#8]
- The Machine Learning Tsunami
  - Machine Learning in Your Projects
  - Objective and Approach
  - Code Examples
  - Prerequisites
  - Roadmap
  - Changes Between the First and the Second Edition
  - Changes Between the Second and the Third Edition
  - Other Resources
  - Conventions Used in This Book
  - O'Reilly Online Learning
  - How to Contact Us
  - Acknowledgments
- I. The Fundamentals of Machine Learning
- 1. The Machine Learning Landscape
  - What Is Machine Learning?
  - Why Use Machine Learning?
  - Examples of Applications
  - Types of Machine Learning Systems
  - Training Supervision
  - Main Challenges of Machine Learning
  - Insufficient Quantity of Training Data
  - Testing and Validating
  - Hyperparameter Tuning and Model Selection
  - Exercises
- 2. End-to-End Machine Learning Project
  - Working with Real Data
  - Look at the Big Picture
  - Frame the Problem
  - Get the Data
  - Running the Code Examples Using Google Colab
  - Explore and Visualize the Data to Gain Insights
  - Visualizing Geographical Data
  - Prepare the Data for Machine Learning Algorithms
  - Clean the Data
  - Select and Train a Model
  - Train and Evaluate on the Training Set
  - Fine-Tune Your Model
  - Grid Search
  - Launch, Monitor, and Maintain Your System
  - Try It Out!
  - Exercises
- 3. Classification
  - MNIST
  - Training a Binary Classifier
  - Performance Measures
  - Measuring Accuracy Using Cross-Validation
  - Multiclass Classification
  - Error Analysis
  - Multilabel Classification
  - Multioutput Classification
  - Exercises
- 4. Training Models
  - Linear Regression
  - The Normal Equation
  - Gradient Descent
  - Batch Gradient Descent
  - Polynomial Regression
  - Learning Curves
  - Regularized Linear Models
  - Ridge Regression
  - Logistic Regression
  - Estimating Probabilities
  - Exercises
- 5. Support Vector Machines
  - Linear SVM Classification
  - Soft Margin Classification
  - Nonlinear SVM Classification
  - Polynomial Kernel
  - SVM Regression
  - Under the Hood of Linear SVM Classifiers
  - The Dual Problem
  - Kernelized SVMs
  - Exercises
- 6. Decision Trees
  - Training and Visualizing a Decision Tree
  - Making Predictions
  - Estimating Class Probabilities
  - The CART Training Algorithm
  - Computational Complexity
  - Gini Impurity or Entropy?
  - Regularization Hyperparameters
  - Regression
  - Sensitivity to Axis Orientation
  - Decision Trees Have a High Variance
  - Exercises
- 7. Ensemble Learning and Random Forests
  - Voting Classifiers
  - Bagging and Pasting
  - Bagging and Pasting in Scikit-Learn
  - Random Forests
  - Extra-Trees
  - Boosting
  - AdaBoost
  - Stacking
  - Exercises
- 8. Dimensionality Reduction
  - The Curse of Dimensionality
  - Main Approaches for Dimensionality Reduction
  - Projection
  - PCA
  - Preserving the Variance
  - Random Projection
  - LLE
  - Other Dimensionality Reduction Techniques
  - Exercises
- 9. Unsupervised Learning Techniques
  - Clustering Algorithms: k-means and DBSCAN
  - k-means
  - Gaussian Mixtures
  - Using Gaussian Mixtures for Anomaly Detection
  - Exercises
- II. Neural Networks and Deep Learning
- 10. Introduction to Artificial Neural Networks with Keras
  - From Biological to Artificial Neurons
  - Biological Neurons
  - Implementing MLPs with Keras
  - Building an Image Classifier Using the Sequential API
  - Fine-Tuning Neural Network Hyperparameters
  - Number of Hidden Layers
  - Exercises
- 11. Training Deep Neural Networks
  - The Vanishing/Exploding Gradients Problems
  - Glorot and He Initialization
  - Reusing Pretrained Layers
  - Transfer Learning with Keras
  - Faster Optimizers
  - Momentum
  - Learning Rate Scheduling
  - Avoiding Overfitting Through Regularization
  - ℓ1 and ℓ2 Regularization
  - Summary and Practical Guidelines
  - Exercises
- 12. Custom Models and Training with TensorFlow
  - A Quick Tour of TensorFlow
  - Using TensorFlow like NumPy
  - Tensors and Operations
  - Customizing Models and Training Algorithms
  - Custom Loss Functions
  - TensorFlow Functions and Graphs
  - AutoGraph and Tracing
  - Exercises
- 13. Loading and Preprocessing Data with TensorFlow
  - The tf.data API
  - Chaining Transformations
  - The TFRecord Format
  - Compressed TFRecord Files
  - Keras Preprocessing Layers
  - The Normalization Layer
  - The TensorFlow Datasets Project
  - Exercises
- 14. Deep Computer Vision Using Convolutional Neural Networks
  - The Architecture of the Visual Cortex
  - Convolutional Layers
  - Filters
  - Pooling Layers
  - Implementing Pooling Layers with Keras
  - CNN Architectures
  - LeNet-5
  - Implementing a ResNet-34 CNN Using Keras
  - Using Pretrained Models from Keras
  - Pretrained Models for Transfer Learning
  - Classification and Localization
  - Object Detection
  - Fully Convolutional Networks
  - Object Tracking
  - Semantic Segmentation
  - Exercises
- 15. Processing Sequences Using RNNs and CNNs
  - Recurrent Neurons and Layers
  - Memory Cells
  - Training RNNs
  - Forecasting a Time Series
  - The ARMA Model Family
  - Handling Long Sequences
  - Fighting the Unstable Gradients Problem
  - Exercises
- 16. Natural Language Processing with RNNs and Attention
  - Generating Shakespearean Text Using a Character RNN
  - Creating the Training Dataset
  - Sentiment Analysis
  - Masking
  - An Encoder–Decoder Network for Neural Machine Translation
  - Bidirectional RNNs
  - Attention Mechanisms
  - Attention Is All You Need: The Original Transformer Architecture
  - An Avalanche of Transformer Models
  - Vision Transformers
  - Hugging Face's Transformers Library
  - Exercises
- 17. Autoencoders, GANs, and Diffusion Models
- Efficient Data Representations
  - Performing PCA with an Undercomplete Linear Autoencoder
  - Stacked Autoencoders
  - Implementing a Stacked Autoencoder Using Keras
  - Convolutional Autoencoders
  - Denoising Autoencoders
  - Sparse Autoencoders
  - Variational Autoencoders
  - Generating Fashion MNIST Images
  - Generative Adversarial Networks
  - The Difficulties of Training GANs
  - Diffusion Models
  - Exercises
- 18. Reinforcement Learning
  - Learning to Optimize Rewards
  - Policy Search
  - Introduction to OpenAI Gym
  - Neural Network Policies
  - Evaluating Actions: The Credit Assignment Problem
  - Policy Gradients
  - Markov Decision Processes
  - Temporal Difference Learning
  - Q-Learning
  - Exploration Policies
  - Implementing Deep Q-Learning
  - Deep Q-Learning Variants
  - Fixed Q-value Targets
  - Overview of Some Popular RL Algorithms
  - Exercises
- 19. Training and Deploying TensorFlow Models at Scale
  - Serving a TensorFlow Model
  - Using TensorFlow Serving
  - Deploying a Model to a Mobile or Embedded Device
  - Running a Model in a Web Page
  - Using GPUs to Speed Up Computations
  - Getting Your Own GPU
  - Training Models Across Multiple Devices
  - Model Parallelism
  - Exercises
  - Thank You!
